app/controllers/orders_controller.rb

```diff
@@ -114,6 +114,24 @@ def cancel
     redirect_to checkout_order_path, alert: 'Payment was cancelled'
   end
 
+  def cancel_order
+    if @order.deals.any? { |deal| deal.expiring_soon? }
```

Comment:
can we have a scope named expiring soon instead?

Reply:
done